### PR TITLE
feat: retrieve full doc or neighboring chunks via MCP

### PR DESCRIPTION
**Pull request by rob-9**
## Description

Addressing [#2090](https://github.com/topoteretes/cognee/issues/2090). 

This PR enriches search results with parent document metadata by querying the graph database after vector search, and adds two new MCP tools for retrieving full documents and neighboring chunks.

## Changes
- `chunks_retriever.py`: After vector search, queries the graph DB for `is_part_of` relationships to attach `parent_document (id, name, type)` to each chunk's payload. Uses batched query with fallback to individual queries. Adds `strict_enrichment` parameter to control error handling.

    I decided to query the graph DB at search time because:                      
    1. The `is_part_of` relationship already exists in the graph, no need to denormalize into vector DB                                 
    2. Adding primitives to `DocumentChunk` model would force users to re-cognify data                        
    3. Keeps vector DB handling embeddings and graph DB handling relationships.

    Tradeoff: one extra batched graph query per search.

- `cognee-mcp/src/server.py`: Added `get_document` tool (retrieve full document with all chunks) and `get_chunk_neighbors` tool (retrieve surrounding chunks by `chunk_index`). 
- `chunks_retriever_test.py`: Added 20 unit tests covering all new code paths (29 total). 
- `cognee-mcp/README.md`: Documented new MCP tools.

Only `CHUNKS` search type returns structured payloads where `parent_document.id` is exposed. The agent uses that ID to access `get_document()` and `get_chunk_neighbors()` tools. Not applicable to `RAG_COMPLETION` (returns pure LLM string response, chunk metadata lost) or `GRAPH_COMPLETION` (operates on entity triplets) right now. 

Tools are still usable alongside any search type. The agent just needs a `CHUNKS` search to obtain IDs. 

^^ Correct me if I'm wrong. New here.

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Code refactoring
- [ ] Other (please specify):

## Screenshots

Screenshot 2026-02-11 at 3 54 01 PM

Need updates to `cognee/tests/integration/retrieval/test_chunks_retriever.py`?

## Pre-submission Checklist

- [x] **I have tested my changes thoroughly before submitting this PR** (See `CONTRIBUTING.md`)
- [x] **This PR contains minimal changes necessary to address the issue/feature**
- [x] My code follows the project's coding standards and style guidelines
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if applicable)
- [x] All new and existing tests pass
- [x] I have searched existing PRs to ensure this change hasn't been submitted already
- [x] I have linked any relevant issues in the description
- [x] My commits have clear and descriptive messages

## DCO Affirmation
I affirm that all code in every commit of this pull request conforms to the terms of the Topoteretes Developer Certificate of Origin.

## Summary by CodeRabbit

* **New Features**
  * Retrieve complete documents with all chunks and optional metadata.
  * Retrieve surrounding chunks (neighboring context) with options to include the target chunk.
  * Retrieved chunks are automatically enriched with parent-document metadata.

* **Documentation**
  * Added "Document Retrieval Examples" and updated API-mode limitations and public tool listings.

* **Tests**
  * Expanded coverage for enrichment, fallbacks, strict-mode behavior, and edge cases.

### CODE REVIEW

**Comment by rob-9** [comment]
rebased onto dev

File: cognee-mcp/src/server.py [hunk 1]
```diff
@@ -517,6 +517,343 @@ async def search_task(search_query: str, search_type: str, top_k: int) -> str:
...
+                """
+
+                result = await graph_engine.query(query, params={"doc_id": document_id})
+
+                if not result or not result[0].get("doc"):
+                    return json.dumps(
+                        {"error": "Document not found", "document_id": document_id}, cls=JSONEncoder
+                    )
+
+                doc = result[0]["doc"]
```
[the remark this answers is not in the document]
Comment (rob-9) [reply cognee-mcp/src/server.py:618]: @Vasilije1990 should I add a helper util function for this or change kuzu adapter query() implementaiton to return type `List[Dict[str, Any]]`?
